app/src/main/java/org/simple/clinic/user/UserSession.kt

```diff
@@ -118,11 +121,20 @@ class UserSession @Inject constructor(
         .doOnSuccess { Timber.i("Login result: $it") }
   }
 
+  private fun reportUserLoggedInToAnalytics() {
+    loggedInUser()
+        .subscribeOn(schedulersProvider.io())
+        .take(1)
+        .filterAndUnwrapJust()
+        .map(Analytics::setLoggedInUser)
+        .subscribe()
```

Comment:
I do understand that `take(1)` should handle unsubscription, but a dangling `.subscribe()` looks suspicious.

Reply:
I agree. This is something that is a side-effect and should happen regardless if the source stream was disposed or not.

However, this could be overkill on my part, and I could just return a `Completable` instead and let it be part of the source stream instead of a side effect. What do you think?